packages/app-project/src/components/AuthModal/shared/components/withCustomFormik/withCustomFormik.spec.js

```diff
@@ -60,32 +60,29 @@ describe('Higher Order Component > withCustomFormik', function () {
     expect(formikHOC.props().validate).to.equal(validate)
   })
 
-  it('should call onChange prop', function () {
-    const onChangeStub = sinon.stub()
+  it('should call the custom change handler on change', function () {
 
+    const onChangeStub = sinon.stub()
     const wrapper = mount(
       <WrappedMockForm
         initialValues={initialValues}
         onChange={onChangeStub}
         onSubmit={onSubmit}
+        ref={mockForm}
       />
     )
-    const formikProps = wrapper.find(MockForm).props()
-    let testInput = wrapper.find('input[name="testInput"]')
+    const mockForm = wrapper.find(MockForm)
+    const formikProps = mockForm.props()
     const eventMock = {
       target: {
-        name: testInput.props().name,
+        name: 'testInput',
         value: 'foo'
       }
     }
+    const changeSpy = onChangeStub.callsFake((event, props) => console.log(event === eventMock, props === formikProps))
 
-    expect(testInput.props().value).to.equal('')
-    testInput.simulate('change', eventMock)
-    // diry chai does not work with sinon chai
-    expect(onChangeStub.withArgs(eventMock, formikProps)).to.have.been.calledOnce
-    testInput = wrapper.find('input[name="testInput"]')
-    // Formik's handleChange updates the input value for us based on the name attribute
-    // Test that the internal Formik handleChange has been called and worked
-    expect(testInput.props().value).to.equal('foo')
+    formikProps.handleChange(eventMock, formikProps)
```

Comment:
Will this set up some weird recursive behaviour because the custom handler function calls `formikProps.handleChange`?

Reply:
Maybe this is why the test fails, because that function has been modified so doesn't match?

Reply:
Yes, this test is failing because we should be testing that the handler is called with the forms original props (`innerProps` in the code), not the modified props.